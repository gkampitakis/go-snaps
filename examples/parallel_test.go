package examples

import (
	"bytes"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestParallel(t *testing.T) {
	type testCases struct {
		description string
		input       any
	}

	value := 10

	tests := []testCases{
		{
			description: "should snap an integer",
			input:       10,
		},
		{
			description: "should snap a float",
			input:       float64(10.5),
		},
		{
			description: "should snap a struct",
			input: struct {
				user  string
				email string
				age   int
			}{
				"gkampitakis",
				"mock@mail.com",
				10,
			},
		},
		{
			description: "should snap a struct with fields",
			input: struct {
				_    struct{}
				name string
				id   string
			}{
				name: "mock-name",
				id:   "123456",
			},
		},
		{
			description: "should snap an integer slice",
			input:       []int{1, 2, 3, 4},
		},
		{
			description: "should snap a map",
			input: map[string]int{
				"value-0": 0,
				"value-1": 1,
				"value-2": 2,
				"value-3": 3,
			},
		},
		{
			description: "should snap a buffer",
			input:       bytes.NewBufferString("Buffer string"),
		},
		{
			description: "should snap a pointer",
			input:       &value,
		},
	}

	for _, scenario := range tests {
		// capture range variable
		s := scenario

		t.Run(s.description, func(t *testing.T) {
			t.Parallel()

			snaps.MatchSnapshot(t, s.input)
		})
	}
}
