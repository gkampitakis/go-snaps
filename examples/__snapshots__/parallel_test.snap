
[TestParallel/should_snap_an_integer - 1]
int(10)
---

[TestParallel/should_snap_a_struct_with_fields - 1]
struct { _ struct {}; name string; id string }{
    _:    struct {}{},
    name: "mock-name",
    id:   "123456",
}
---

[TestParallel/should_snap_an_integer_slice - 1]
[]int{1, 2, 3, 4}
---

[TestParallel/should_snap_a_float - 1]
float64(10.5)
---

[TestParallel/should_snap_a_map - 1]
map[string]int{"value-0":0, "value-1":1, "value-2":2, "value-3":3}
---

[TestParallel/should_snap_a_buffer - 1]
&bytes.Buffer{
    buf:      {0x42, 0x75, 0x66, 0x66, 0x65, 0x72, 0x20, 0x73, 0x74, 0x72, 0x69, 0x6e, 0x67},
    off:      0,
    lastRead: 0,
}
---

[TestParallel/should_snap_a_pointer - 1]
&int(10)
---

[TestParallel/should_snap_a_struct - 1]
struct { user string; email string; age int }{user:"gkampitakis", email:"mock@mail.com", age:10}
---
